Sources/SourceKitLSP/Swift/FoldingRange.swift

```diff
@@ -264,3 +277,14 @@ extension SwiftLanguageService {
     return ranges.sorted()
   }
 }
+
+fileprivate extension LineTable {
+  func isAtEndOfLine(_ position: Position) -> Bool {
+    guard position.line >= 0, position.line < self.count else {
+      return false
+    }
+    let line = self[position.line]
+    let suffixAfterPositionColumn = line[line.utf16.index(line.startIndex, offsetBy: position.utf16index)...]
+    return suffixAfterPositionColumn.allSatisfy(\.isNewline)
```

Comment:
Should we care about trailing whitespace too?

Reply:
I don’t think it matters in practice because this is really only for comment folding, which would include the whitespace in the comment’s trivia and thus point to the position right before the newline. 

Let’s leave it for now because I don’t have a clean name for this function if we also consider whitespaces.